### ISSUE TITLE
Livechat: Incoming Livechats not disappearing

### ISSUE THREAD

**Issue by NIbbel92** [RocketChat/Rocket.Chat#15748]
We are having trouble removing 2 incoming livechats.
It is not possible to hide/delete the Rooms.

Till now we could not reproduce this issue.

 Server Setup Information:

- Version of Rocket.Chat Server: 2.2.0
- Operating System: CentOS 7
- Deployment Method: tar
- Number of Running Instances: 1
- DB Replicaset Oplog: yes, activated
- NodeJS Version: 8.12.0
- MongoDB Version: 4.0.13

Client Setup Information

- Desktop App or Browser Version: App
- Operating System: Mac OS 10.14.6

If any further information is needed please let me know.

Best Nibbel

**Comment by rehste** [comment]
Hello together,

i have the same problem... 👎

**Comment by renatobecker** [comment]
Hi @NIbbel92, can you please record a video screen or something?
Thanks.

**Comment by NIbbel92** [comment]
Video is a bit problematic, but i have attached a screenshot.
This happens when trying to enter the incoming chat.
![rocketchat_livechat](https://user-images.githubusercontent.com/57345650/68300953-2d6ade80-009f-11ea-911a-7ae41216fb07.png)

**Comment by renatobecker** [comment]
Hi, @NIbbel92 are you sure you were running the `2.2.0` version when this happened?
I'm asking because we were aware of it and it's supposed to be fixed after `2.2.0` release.

**Comment by rehste** [comment]
Hi @renatobecker 
I've had this problem for a few months now. I can't close my chats either, because the error comes "Room not found". So the chats didn't come after version 2.2.0 but we somehow need a solution to delete the chats.

**Comment by renatobecker** [comment]
> Hi @renatobecker
> I've had this problem for a few months now. I can't close my chats either, because the error comes "Room not found". So the chats didn't come after version 2.2.0 but we somehow need a solution to delete the chats.

Do you have access to the database?
If so, I can guide you on how to fix it.

**Comment by rehste** [comment]
> Do you have access to the database?
> If so, I can guide you on how to fix it.

@renatobecker yes. Please tell me where to find the entries, then I will delete them. Thank you

**Comment by renatobecker** [comment]
> > Do you have access to the database?
> > If so, I can guide you on how to fix it.
> 
> @renatobecker yes. Please tell me where to find the entries, then I will delete them. Thank you

This command should solve the issue:
`db.rocketchat_livechat_inquiry.update({status: 'queued',closedBy: {$exists: 1}},{$set: {status: ‘closed’}},{multi: true})
`

**Comment by rehste** [comment]
@renatobecker 

with: 

> db.rocketchat_livechat_inquiry.update({status: 'queued',closedBy: {$exists: 1}},{$set: {status: 
> ‘closed’}},{multi: true}) 

get the following error:

`2019-11-11T21:20:51.537+0100 E QUERY    [js] SyntaxError: illegal character @(shell):1:96
rs01:PRIMARY> db.rocketchat_livechat_inquiry.update({status: 'queued',closedBy: {$exists: 1}},{$set: {status: ‘closed’}},{multi: true})
2019-11-11T21:21:19.567+0100 E QUERY    [js] SyntaxError: illegal character @(shell):1:96
rs01:PRIMARY>`

**Comment by renatobecker** [comment]
> @renatobecker
> 
> with:
> 
> > db.rocketchat_livechat_inquiry.update({status: 'queued',closedBy: {$exists: 1}},{$set: {status:
> > ‘closed’}},{multi: true})
> 
> get the following error:
> 
> `2019-11-11T21:20:51.537+0100 E QUERY [js] SyntaxError: illegal character @(shell):1:96 rs01:PRIMARY> db.rocketchat_livechat_inquiry.update({status: 'queued',closedBy: {$exists: 1}},{$set: {status: ‘closed’}},{multi: true}) 2019-11-11T21:21:19.567+0100 E QUERY [js] SyntaxError: illegal character @(shell):1:96 rs01:PRIMARY>`

It seems to be a problem with the quoted string `closed`. Please, try the command below:
`db.rocketchat_livechat_inquiry.update({status: 'queued',closedBy: {$exists: 1}},{$set: {status: 'closed'}},{multi: true})`

**Comment by rehste** [comment]
`rs01:PRIMARY> db.rocketchat_livechat_inquiry.update({status: 'queued',closedBy: {$exists: 1}},{$set: {status: 'closed'}},{multi: true})
WriteResult({ "nMatched" : 0, "nUpserted" : 0, "nModified" : 0 })`

Unfortunately the two incoming live chats, which we can't delete, are still there.

**Comment by renatobecker** [comment]
> `rs01:PRIMARY> db.rocketchat_livechat_inquiry.update({status: 'queued',closedBy: {$exists: 1}},{$set: {status: 'closed'}},{multi: true}) WriteResult({ "nMatched" : 0, "nUpserted" : 0, "nModified" : 0 })`
> 
> Unfortunately the two incoming live chats, which we can't delete, are still there.

In this case, you'll need to update the documents passing the `rid` parameter through the query. For example:

`db.rocketchat_livechat_inquiry.update({rid: 'ZqCQbYjCQbqXZesqY'},{$set: {status: 'closed'}})`

You can get the `rid` value from the `URL`, so you'll need to call this command for each `inquiry`.

**Comment by rehste** [comment]
I went into Rocket.chat then clicked "open link" and copied the rid at the end of the url. Unfortunately no status was changed to "closed". 

<img width="780" alt="Bildschirmfoto 2019-11-11 um 22 41 15" src="https://user-images.githubusercontent.com/7567967/68623274-64dcef00-04d4-11ea-92f3-bb7b73f697ae.png">

**Comment by renatobecker** [comment]
> I went into Rocket.chat then clicked "open link" and copied the rid at the end of the url. Unfortunately no status was changed to "closed".
> 
> <img alt="Bildschirmfoto 2019-11-11 um 22 41 15" width="780" src="https://user-images.githubusercontent.com/7567967/68623274-64dcef00-04d4-11ea-92f3-bb7b73f697ae.png">

This is really weird. What happens when you find the inquiry by the `rid`? Do you get any results?

**Comment by rehste** [comment]
@renatobecker now it works.. Thank you :)

**Comment by NIbbel92** [comment]
> Hi, @NIbbel92 are you sure you were running the `2.2.0` version when this happened?
> I'm asking because we were aware of it and it's supposed to be fixed after `2.2.0` release.

yes the issue was present prior upgrading to 2.2.0
I have followed your. steps on deleting the chats directly in the database.
Chats are gone now thank you.
Ticket can be closed

**Comment by powed** [comment]
> > I went into Rocket.chat then clicked "open link" and copied the rid at the end of the url. Unfortunately no status was changed to "closed".
> > <img alt="Bildschirmfoto 2019-11-11 um 22 41 15" width="780" src="https://user-images.githubusercontent.com/7567967/68623274-64dcef00-04d4-11ea-92f3-bb7b73f697ae.png">
> 
> This is really weird. What happens when you find the inquiry by the `rid`? Do you get any results?

Hello, we have the same problem.

**Comment by exequos** [comment]
Hello, We have the some problem after upgrade.
There are Livechats with "Room not found" which cannot be closed neither removed by suggested db.rocketchat_livechat_inquiry.update({rid: '<the rid value from the URL>'},{$set: {status: 'closed'}})

Is there any other solution?


Version | 2.4.0
Mongo Version | 3.6.14

**Comment by danielboydston** [comment]
I recently experience this issue as well.  I'm unsure if the issue existed before we upgraded or not, but we went from 0.? to 2.4.9, after which, the issue was brought to my attention.

I was able to resolve our issue by using the following commands in the mongo shell (the first query suggestion from renatobecker-zz:
`use rocketchat
db.rocketchat_livechat_inquiry.update({status: 'queued',closedBy: {$exists: 1}},{$set: {status: 'closed'}},{multi: true})`

For those of us not familiar with MongoDB, if the "use" statement is omitted, the query will not modify any records, therefore not solving the problem because it is not operating on the correct database.  This fouled me up at first until I figured this out.
